### ISSUE TITLE
Inconsistent action dimension when using target_rotations

### ISSUE THREAD

**Issue by summertaiyuan** [NVIDIA/Isaac-GR00T#282]
### 🐛 Describe the bug

When using `StateActionTransform` with `target_rotations={"action.axis_angle": "quaternion"}`, the action dimension changes (e.g., from 3 for axis-angle to 4 for quaternion), but `self.action_dims` does not get updated accordingly.

This causes the following assertion to fail:

```python
assert (
    self.action_dims[key] == data[key].shape[-1]
), f"Action dim mismatch for {key=}, {self.action_dims[key]=}, {data[key].shape[-1]=}"
```

Because the expected action dimension does not match the transformed data’s last dimension.

This PR highlights the dimension mismatch issue that needs to be addressed by updating `self.action_dims` when applying rotation transformations.

### Versions

absl-py==2.3.0
accelerate==1.2.1
action-msgs==1.2.1
action-tutorials-interfaces==0.20.5
action-tutorials-py==0.20.5
actionlib-msgs==4.8.0
aiosignal==1.3.2
albucore==0.0.17
albumentations==1.4.18
ament-cmake-test==1.3.11
ament-copyright==0.12.12
ament-cppcheck==0.12.12
ament-cpplint==0.12.12
ament-flake8==0.12.12
ament-index-python==1.4.0
ament-lint==0.12.12
ament-lint-cmake==0.12.12
ament-package==0.14.0
ament-pep257==0.12.12
ament-uncrustify==0.12.12
ament-xmllint==0.12.12
angles==1.15.0
annotated-types==0.7.0
antlr4-python3-runtime==4.9.3
anyio==4.9.0
argon2-cffi==25.1.0
argon2-cffi-bindings==21.2.0
arm_pytorch_utilities==0.4.3
arrow==1.3.0
asttokens==3.0.0
astunparse==1.6.3
async-lru==2.0.5
attrs==25.3.0
av==12.3.0
babel==2.17.0
beautifulsoup4==4.13.4
bleach==6.2.0
blessings==1.7
builtin-interfaces==1.2.1
cachetools==5.5.2
certifi==2025.6.15
cffi==1.17.1
charset-normalizer==3.4.2
click==8.2.1
cloudpickle==3.1.1
comm==0.2.2
composition-interfaces==1.2.1
contourpy==1.3.2
control-msgs==4.8.0
controller-manager==2.50.0
controller-manager-msgs==2.50.0
cramjam==2.10.0
cv-bridge==3.2.1
cycler==0.12.1
dacite==1.9.2
debugpy==1.8.14
decorator==5.2.1
decord==0.6.0
defusedxml==0.7.1
demo-nodes-py==0.20.5
diagnostic-msgs==4.8.0
diagnostic-updater==4.0.4
diffusers==0.30.2
dm-tree==0.1.8
docker-pycreds==0.4.0
docstring_parser==0.16
domain-coordinator==0.10.0
einops==0.8.1
eval_type_backport==0.2.2
example-interfaces==0.9.3
examples-rclpy-executors==0.15.3
examples-rclpy-minimal-action-client==0.15.3
examples-rclpy-minimal-action-server==0.15.3
examples-rclpy-minimal-client==0.15.3
examples-rclpy-minimal-publisher==0.15.3
examples-rclpy-minimal-service==0.15.3
examples-rclpy-minimal-subscriber==0.15.3
exceptiongroup==1.3.0
executing==2.2.0
Farama-Notifications==0.0.4
fast_kinematics==0.2.2
fastjsonschema==2.21.1
fastparquet==2024.11.0
ffmpeg-image-transport-msgs==1.1.2
filelock==3.18.0
flash-attn==2.7.1.post4
flatbuffers==25.2.10
fonttools==4.58.4
fqdn==1.5.1
frozenlist==1.7.0
fsspec==2025.5.1
fvcore==0.1.5.post20221221
gast==0.6.0
geometry-msgs==4.8.0
gitdb==4.0.12
GitPython==3.1.44
google-auth==2.40.3
google-auth-oauthlib==1.2.2
google-pasta==0.2.0
-e git+https://github.com/NVIDIA/Isaac-GR00T.git@7762576bbc0c87b2d78e6c5cb0188c46e47d8e66#egg=gr00t
grpcio==1.73.0
gymnasium==0.29.1
h11==0.16.0
h5py==3.12.1
hf-xet==1.1.5
httpcore==1.0.9
httpx==0.28.1
huggingface-hub==0.33.0
hydra-core==1.3.2
idna==3.10
image-geometry==3.2.1
imageio==2.34.2
imageio-ffmpeg==0.6.0
importlib_metadata==8.7.0
iniconfig==2.1.0
interactive-markers==2.3.2
iopath==0.1.10
ipykernel==6.29.5
ipython==8.37.0
isoduration==20.11.0
jedi==0.19.2
Jinja2==3.1.6
joint-state-publisher==2.4.0
joint-state-publisher-gui==2.4.0
json5==0.12.0
jsonpointer==3.0.0
jsonschema==4.24.0
jsonschema-specifications==2025.4.1
jupyter-events==0.12.0
jupyter-lsp==2.2.5
jupyter_client==8.6.3
jupyter_core==5.8.1
jupyter_server==2.16.0
jupyter_server_terminals==0.5.3
jupyterlab==4.4.4
jupyterlab_pygments==0.3.0
jupyterlab_server==2.27.3
keras==2.15.0
kiwisolver==1.4.8
kornia==0.7.4
kornia_rs==0.1.9
laser-geometry==2.4.0
launch==1.0.8
launch-ros==0.19.9
launch-testing==1.0.8
launch-testing-ros==0.19.9
launch-xml==1.0.8
launch-yaml==1.0.8
lazy_loader==0.4
libclang==18.1.1
lifecycle-msgs==1.2.1
llvmlite==0.44.0
logging-demo==0.20.5
lxml==6.0.0
# Editable install with no version control (mani_skill==3.0.0b21)
-e /home/gaoteng/workspace/ManiSkill-main
map-msgs==2.1.0
Markdown==3.8.2
markdown-it-py==3.0.0
MarkupSafe==3.0.2
matplotlib==3.10.0
matplotlib-inline==0.1.7
mdurl==0.1.2
message-filters==4.3.7
mistune==3.1.3
ml-dtypes==0.2.0
mplib==0.1.1
mpmath==1.3.0
msgpack==1.1.1
nav-msgs==4.8.0
nbclient==0.10.2
nbconvert==7.16.6
nbformat==5.10.4
nest-asyncio==1.6.0
networkx==3.4.2
notebook==7.4.4
notebook_shim==0.2.4
numba==0.61.2
numpy==1.26.4
numpydantic==1.6.7
nvidia-cublas-cu12==12.4.5.8
nvidia-cuda-cupti-cu12==12.4.127
nvidia-cuda-nvrtc-cu12==12.4.127
nvidia-cuda-runtime-cu12==12.4.127
nvidia-cudnn-cu12==9.1.0.70
nvidia-cufft-cu12==11.2.1.3
nvidia-curand-cu12==10.3.5.147
nvidia-cusolver-cu12==11.6.1.9
nvidia-cusparse-cu12==12.3.1.170
nvidia-ml-py==12.575.51
nvidia-nccl-cu12==2.21.5
nvidia-nvjitlink-cu12==12.4.127
nvidia-nvtx-cu12==12.4.127
oauthlib==3.3.1
omegaconf==2.3.0
onnx==1.15.0
opencv-python==4.8.0.74
opencv-python-headless==4.11.0.86
opt_einsum==3.4.0
osrf-pycommon==2.1.6
overrides==7.7.0
packaging==25.0
pandas==2.2.3
pandocfilters==1.5.1
parso==0.8.4
pcl-msgs==1.0.0
peft==0.14.0
pendulum-msgs==0.20.5
pettingzoo==1.25.0
pexpect==4.9.0
pillow==11.2.1
pipablepytorch3d==0.7.6
platformdirs==4.3.8
pluggy==1.6.0
polars==1.31.0
portalocker==3.2.0
prometheus_client==0.22.1
prompt_toolkit==3.0.51
protobuf==3.20.3
psutil==7.0.0
ptyprocess==0.7.0
pure_eval==0.2.3
pyarrow==14.0.1
pyasn1==0.6.1
pyasn1_modules==0.4.2
pycparser==2.22
pydantic==2.10.6
pydantic_core==2.27.2
Pygments==2.19.2
pynvml==12.0.0
pyparsing==3.2.3
pyperclip==1.9.0
pytest==8.4.1
python-dateutil==2.9.0.post0
python-json-logger==3.3.0
python-qt-binding==1.1.2
pytorch-kinematics==0.7.5
pytorch-seed==0.2.0
pytz==2025.2
PyYAML==6.0.2
pyzmq==27.0.0
qt-dotgraph==2.2.4
qt-gui==2.2.4
qt-gui-cpp==2.2.4
qt-gui-py-common==2.2.4
quality-of-service-demo-py==0.20.5
ray==2.40.0
rcl-interfaces==1.2.1
rclpy==3.3.16
rcutils==5.1.6
referencing==0.36.2
regex==2024.11.6
requests==2.32.3
requests-oauthlib==2.0.0
resource-retriever==3.1.3
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==14.0.0
rmw-dds-common==1.6.0
ros2action==0.18.12
ros2bag==0.15.14
ros2cli==0.18.12
ros2component==0.18.12
ros2doctor==0.18.12
ros2interface==0.18.12
ros2launch==0.19.9
ros2lifecycle==0.18.12
ros2multicast==0.18.12
ros2node==0.18.12
ros2param==0.18.12
ros2pkg==0.18.12
ros2run==0.18.12
ros2service==0.18.12
ros2topic==0.18.12
rosbag2-interfaces==0.15.14
rosbag2-py==0.15.14
rosgraph-msgs==1.2.1
rosidl-adapter==3.1.6
rosidl-cli==3.1.6
rosidl-cmake==3.1.6
rosidl-generator-c==3.1.6
rosidl-generator-cpp==3.1.6
rosidl-generator-py==0.14.4
rosidl-parser==3.1.6
rosidl-runtime-py==0.9.3
rosidl-typesupport-c==2.0.2
rosidl-typesupport-cpp==2.0.2
rosidl-typesupport-fastrtps-c==2.2.2
rosidl-typesupport-fastrtps-cpp==2.2.2
rosidl-typesupport-introspection-c==3.1.6
rosidl-typesupport-introspection-cpp==3.1.6
rpds-py==0.25.1
rpyutils==0.2.1
rqt-action==2.0.1
rqt-bag==1.1.5
rqt-bag-plugins==1.1.5
rqt-console==2.0.3
rqt-graph==1.3.1
rqt-gui==1.1.7
rqt-gui-py==1.1.7
rqt-msg==1.2.0
rqt-plot==1.1.5
rqt-publisher==1.5.0
rqt-py-common==1.1.7
rqt-py-console==1.0.2
rqt-reconfigure==1.1.2
rqt-service-caller==1.0.5
rqt-shell==1.0.2
rqt-srv==1.0.3
rqt-topic==1.5.0
rsa==4.9.1
safetensors==0.5.3
sapien==3.0.0b1
scikit-image==0.25.2
scipy==1.15.3
Send2Trash==1.8.3
sensor-msgs==4.8.0
sensor-msgs-py==4.8.0
sentry-sdk==2.31.0
setproctitle==1.3.6
shape-msgs==4.8.0
shtab==1.7.2
six==1.17.0
smmap==5.0.2
sniffio==1.3.1
soupsieve==2.7
sros2==0.10.6
stack-data==0.6.3
statistics-msgs==1.2.1
std-msgs==4.8.0
std-srvs==4.8.0
stereo-msgs==4.8.0
sympy==1.13.1
tabulate==0.9.0
teleop-twist-keyboard==2.4.0
tensorboard==2.15.2
tensorboard-data-server==0.7.2
tensorflow==2.15.0
tensorflow-estimator==2.15.0
tensorflow-io-gcs-filesystem==0.37.1
termcolor==3.1.0
terminado==0.18.1
tf2-geometry-msgs==0.25.12
tf2-kdl==0.25.12
tf2-msgs==0.25.12
tf2-py==0.25.12
tf2-ros-py==0.25.12
tf2-tools==0.25.12
theora-image-transport==2.5.3
tianshou==0.5.1
tifffile==2025.5.10
timm==1.0.14
tinycss2==1.4.0
tokenizers==0.21.2
tomli==2.2.1
topic-monitor==0.20.5
toppra==0.6.3
torch 
torchvision==0.20.1
tornado==6.5.1
tqdm==4.67.1
traitlets==5.14.3
trajectory-msgs==4.8.0
transformers==4.51.3
transforms3d==0.4.2
trimesh==4.6.13
triton==3.1.0
turtlesim==1.4.2
typeguard==4.4.2
types-python-dateutil==2.9.0.20250516
typing_extensions==4.12.2
tyro==0.9.17
tzdata==2025.2
unique-identifier-msgs==2.2.1
uri-template==1.3.0
urllib3==2.5.0
visualization-msgs==4.8.0
wandb==0.18.0
wcwidth==0.2.13
webcolors==24.11.1
webencodings==0.5.1
websocket-client==1.8.0
Werkzeug==3.1.3
wrapt==1.14.1
xacro==2.0.13
yacs==0.1.8
zipp==3.23.0
zmq==0.0.0

**Comment by xiaotongc0** [comment]
Since n1.6, the ConcatTransform has been refactored and dissembled to multiple parts. There is no self.action_dims that needs manual change, instead the format of rotation is passed by ActionConfig.format (XYZ_ROT6D / XYZ_ROTVEC / DEFAULT) and processed in StateActionProcessor. There is no dimension mismatch problem.
